cmd/dcs-web/search/query-rewrite.go

```diff
@@ -2,11 +2,52 @@
 package search
 
 import (
-	"fmt"
 	"net/url"
+	"regexp"
 	"strings"
 )
 
+var (
+	start = regexp.MustCompile(`(?i)^\s*(-?(?:filetype|package|pkg|path|file)):(\S+)\s+`)
+	end = regexp.MustCompile(`(?i)\s+(-?(?:filetype|package|pkg|path|file)):(\S+)\s*$`)
+)
+
+func parseFilters(query *url.Values, qstr string, atstart bool) string {
+	q := *query
+
+	var re *regexp.Regexp
+	if atstart {
+		re = start
+	} else {
+		re = end
+	}
+
+	matches := re.FindStringSubmatchIndex(qstr)
+	for matches != nil {
+		// matches is [token_start, token_end, type_start, type_end, value_start, value_end]
```

Comment:
Ah, you’re referring to line 40 and 42, I take it? I think using the length of the match is just fine. Maybe you can give that approach a go when you get around to updating this pull request, so that we can compare the two of them directly instead of just talking about it :).